### PR TITLE
Add quiz view

### PR DESCRIPTION
**Pull request by fungjj92**
## Overview

When the user elects to play the quiz, the user is dropped into a fullscreen quiz outside the framework of the 4 tabs. When the quiz is exited, whether by finishing or closing out, we still want to return to the user to the quiz tab and previous app state. This PR adds the quiz view and implements that behavior.

Connects #40 

### Demo

quizinternal

### Notes

Our root folder is growing with components, so I organized them into a `components` subdirectory. We may want a separate folder for styled-components.

## Testing Instructions

Use the Netlify site and navigate to the quiz page. Click Play and then the "close" exit. See the you're sent back to QuizHome and the About state is maintained

### CODE REVIEW

File: src/app/src/App.js [hunk 1]
```diff
@@ -51,11 +55,14 @@ class App extends Component {
...
         );
+
         return (
             <div className='App'>
                 <GlobalStyle />
+                {isScreensaverVisible && screensaver}
                 {!isScreensaverVisible && idleTimer}
-                {app}
+                {isQuizVisible && quiz}
+                {!isScreensaverVisible && app}
```
Comment (maurizi) [inline src/app/src/App.js]: Consider using a ternary operator here, and grouping the two `!isScreensaverVisible` components, to make the control flow more explicit:
```jsx
<div className='App'>
    <GlobalStyle />
    {isQuizVisible && quiz}
    {isScreensaverVisible ? (
        screensaver
    ) : (
        <>
            {idleTimer}
            {app}
        </>
    )}
</div>
```

File: src/app/src/components/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+`;
+
+const StyledNavbar = styled.div`
+    display: ${props => props.hide};
+`;
+
+const Navbar = props => {
+    return (
+        // unmountOnExit is used to ensure that videos restart when switching
+        // from About tab to another tab and back again
```
Comment (maurizi) [inline src/app/src/components/Navbar.js]: Can we move this comment down a line, so it remains above the line it relates to?

File: src/app/src/components/Quiz.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Flex, Box } from 'rebass';
+import { themeGet } from 'styled-system';
+
+import QuizNavbar from './QuizNavbar';
+import Sidebar from './Sidebar';
+
+const StyledQuiz = styled(Box)``;
```
Comment (maurizi) [inline src/app/src/components/Quiz.js]: There isn't a need to have a named `StyledQuiz` if we aren't styling it yet - we can directly use `Box` for now.

File: src/app/src/components/QuizNavbar.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    width: 100%;
+    height: 50px;
+    background-color: #296882;
+    padding: 0 10px 0 30px;
+`;
+
+const CloseButton = styled(Button)``;
```
Comment (maurizi) [inline src/app/src/components/QuizNavbar.js]: Similar to above, there isn't a need to have a named `CloseButton` if we aren't styling it yet - we can directly use `Button` for now.

**Review by maurizi — CHANGES_REQUESTED** [review]
Great to see the reorganization in https://github.com/azavea/pwd-fishway/pull/59/commits/9f624453c9a22e299e7d53bf6defb6f9ffdab881, that was sorely needed.

Functionally this is good.
I left a few comments re: readability that I'd like to see addressed unless you have strong objections.

File: src/app/src/App.js (hunk 1 above)
Comment (fungjj92) [reply src/app/src/App.js]: Additionally I moved the ternary outside of the return JSX to keep that code cleaner.

File: src/app/src/components/Navbar.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+const StyledNavbar = styled.div`
+    display: ${props => props.hide};
+`;
+
+const Navbar = props => {
+    return (
+        <StyledNavbar hide={props.isQuizVisible ? 'none' : 'visible'}>
+            {/* unmountOnExit is used to ensure that videos restart when
+            switching // from About tab to another tab and back again */}
```
Comment (maurizi) [inline src/app/src/components/Navbar.js]:
```suggestion
            switching from About tab to another tab and back again */}
```

Comment (fungjj92) [reply src/app/src/components/Navbar.js]: 🤦‍♀

**Review by maurizi — APPROVED** [review]
🎉 looks good!

**Comment by fungjj92** [comment]
Thanks for reviews! :D

**Pull request merged**